translator/translate/otel/common/eksdetector.go

```diff
@@ -39,36 +47,41 @@ var (
 	// NewDetector creates a new singleton detector for EKS
 	NewDetector = func() (Detector, error) {
 		var errors error
+		if clientset, err := getClient(); err != nil {
+			errors = err
+		} else {
+			detector = &EksDetector{Clientset: clientset}
+		}
+
+		return detector, errors
+	}
+
+	// IsEKS checks if the agent is running on EKS. This is done by using the kubernetes API to determine if the aws-auth
+	// configmap exists in the kube-system namespace
+	IsEKS = func() IsEKSCache {
 		once.Do(func() {
-			if clientset, err := getClient(); err != nil {
+			var errors error
+			var value bool
+			// Create eks detector
+			eksDetector, err := NewDetector()
+			if err != nil {
 				errors = err
-			} else {
-				detectorSingleton = &EksDetector{Clientset: clientset}
 			}
+
+			if eksDetector != nil {
+				// Make HTTP GET request
+				awsAuth, err := eksDetector.getConfigMap(authConfigNamespace, authConfigConfigMap)
+				if err == nil {
```

Comment:
It does, but in that case IsEKS should "return" false, which it will as false is the default. Same concept as before: 
```
	awsAuth, err := eksDetector.getConfigMap(authConfigNamespace, authConfigConfigMap)
	if err != nil {
		return false, nil
	}
```